homeassistant/helpers/script.py

```diff
@@ -539,6 +549,64 @@ async def _async_choose_step(self):
         if choose_data["default"]:
             await self._async_run_script(choose_data["default"])
 
+    async def _async_wait_for_trigger_step(self):
+        """Wait for a trigger event."""
+        if CONF_TIMEOUT in self._action:
+            delay = self._get_pos_time_period_template(CONF_TIMEOUT).total_seconds()
+        else:
+            delay = None
+
+        self._script.last_action = self._action.get(CONF_ALIAS, "wait for trigger")
+        self._log(
+            "Executing step %s%s",
+            self._script.last_action,
+            "" if delay is None else f" (timeout: {timedelta(seconds=delay)})",
+        )
+
+        variables = deepcopy(self._variables)
+        self._variables["wait"] = {"remaining": delay, "trigger": None}
```

Comment:
We don't need a deepcopy if we're just overriding a top level key.
```suggestion
        variables = {**self._variables, "wait": {"remaining": delay, "trigger": None}}
```

Reply:
oh nevermind, one is for passing to trigger, that's different.

Reply:
It's not overriding `wait`, it's specifically _avoiding_ doing that. The call to `async_initialize_triggers` needs a copy of the variables as they existed when this step started. That's what `variables` is for. It gets a copy before `self._variables` is updated during the execution of the step, and it does it in such a way as to not be affected by that. Hence the deepcopy.

Reply:
I could definitely add a comment to make that more clear.